src/components/profile/UserProfilePage/index.tsx

```diff
@@ -0,0 +1,192 @@
+            </Tooltip>
+          </div>
+          <div className={styles.cardRank}>
+            <div className={styles.rank}>{getUserRank(handleUser.points)}</div>
+            <div className={styles.points}>
+              <LeaderboardIcon /> &nbsp;
+              {handleUser.points.toLocaleString()} All-Time Leaderboard Points
+            </div>
+          </div>
+          {isSignedInUser && (
+            <div className={styles.editWrapper}>
+              <Link href={config.profile.editRoute}>
+                <div>
+                  <EditIcon />
+                </div>
+              </Link>
+            </div>
+          )}
+        </div>
+      </div>
+      <div className={`${styles.section} ${styles.progressSection}`}>
+        <Typography variant="h2/bold">
+          {isSignedInUser ? 'My' : `${handleUser.firstName}'s`} Progress
+        </Typography>
+        <div className={styles.progressInfo}>
+          <Typography variant="h4/regular">Level {getLevel(handleUser.points)}</Typography>
+          <Typography variant="h4/regular">{handleUser.points % 100}/100</Typography>
+          <div className={styles.progressBar}>
+            <div className={styles.inner} style={{ width: `${progress}%` }} />
+          </div>
+        </div>
+        <Typography variant="h5/regular" component="p">
+          {isSignedInUser ? 'You need ' : `${handleUser.firstName} needs `}
+          {100 - (handleUser.points % 100)} more points to level up to
+          <Typography variant="h5/bold" component="span">
+            &nbsp;{getUserRank(handleUser.points + 100)}
+          </Typography>
+        </Typography>
+      </div>
+
+      <div className={`${styles.section} ${styles.aboutSection}`}>
+        <div>
+          <Typography variant="h2/bold">About me</Typography>
+          <div className={styles.aboutMeSection}>
+            <ProfileIcon className={styles.icon} />
+            <Typography variant="h4/regular">Class of {handleUser.graduationYear}</Typography>
+            <MajorIcon className={styles.icon} />
+            <Typography variant="h4/regular">{handleUser.major}</Typography>
+          </div>
+          <div>
+            {handleUser.userSocialMedia &&
+              Object.entries(socialMediaIcons)?.map(([key, Icon]) => {
+                const matchingSocialMedia = handleUser.userSocialMedia.find(
+                  (obj: PublicUserSocialMedia) => obj.type === key
+                );
+                if (!matchingSocialMedia) return null;
+                return (
+                  <a key={key} href={matchingSocialMedia.url}>
+                    <Icon />
+                  </a>
+                );
+              })}
+          </div>
+        </div>
+        <div className={styles.bioSection}>
+          <Typography variant="h2/bold">Bio</Typography>
+          <Typography variant="h5/medium" component="p">
+            {handleUser.bio || <i>Nothing here...</i>}
+          </Typography>
+        </div>
```

Comment:
Maybe you could just hide the entire Bio section if the user bio is empty

Reply:
I think the card looks kinda empty and weird without it so I'm going to leave it for now